src/python/pants/backend/python/lint/python_format_target.py

```diff
@@ -29,13 +31,18 @@ class _ConcretePythonFormatTarget:
 @rule
 async def format_python_target(
   wrapped_target: _ConcretePythonFormatTarget, union_membership: UnionMembership
-) -> FmtResults:
-  """This aggregator allows us to have multiple formatters operate over the same Python targets."""
-  results = await MultiGet(
-    Get[FmtResult](PythonFormatTarget, member(wrapped_target.target))
-    for member in union_membership.union_rules[PythonFormatTarget]
-  )
-  return FmtResults(results)
+) -> AggregatedFmtResults:
+  """This aggregator allows us to have multiple formatters safely operate over the same Python
+  targets, even if they modify the same files."""
+  prior_formatter_result_digest: Optional[Digest] = None
+  results: List[FmtResult] = []
+  for member in union_membership.union_rules[PythonFormatTarget]:
```

Comment:
What happens if no `UnionRule`s are defined for `PythonFormatTarget`?

Reply:
Fixed the implementation per Daniel's recommendation. `combined_digest` will be the original digest for the target, so the core `fmt.py` will no-op when overwriting in the workspace. `results` will be empty, so the core `fmt.py` won't print any stdout or stderr for Python targets.